From 56e9c345ef4e41e883f204f8fcd57d306037a26f Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 27 Jul 2023 08:58:19 +0800
Subject: [PATCH] [riscv] Delete debug check in
 InterpreterPushArgsThenFastConstructFunction

cp (context pointer) register is not pushed by EnterFrame for RISC-V, so  the debug check should be removed.

Change-Id: Id0dfe4cdc4c26ace5ba285744903bfffb8e4e245
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4722771
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#89226}
---
 src/builtins/riscv/builtins-riscv.cc | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/builtins/riscv/builtins-riscv.cc b/src/builtins/riscv/builtins-riscv.cc
index d224fddc33f..f7fc0799ebc 100644
--- a/src/builtins/riscv/builtins-riscv.cc
+++ b/src/builtins/riscv/builtins-riscv.cc
@@ -1568,12 +1568,6 @@ void Builtins::Generate_InterpreterPushArgsThenFastConstructFunction(
   FrameScope scope(masm, StackFrame::MANUAL);
   __ EnterFrame(StackFrame::FAST_CONSTRUCT);
 
-  if (v8_flags.debug_code) {
-    // Check that FrameScope pushed the context on to the stack already.
-    __ LoadReceiver(a2);
-    __ Check(eq, AbortReason::kUnexpectedValue, a2, Operand(cp));
-  }
-
   // Implicit receiver stored in the construct frame.
   __ LoadRoot(a2, RootIndex::kTheHoleValue);
   __ Push(cp, a2);
-- 
2.35.1

